nixos/default.nix

```diff
@@ -169,6 +167,13 @@
     ]
     ++ lib.optionals (service.whoisTimeout != null) ["-whoisTimeout" service.whoisTimeout]
     ++ lib.optionals (service.upstreamUnixAddr != null) ["-upstreamUnixAddr" service.upstreamUnixAddr]
+    ++ lib.optionals (
+      (lib.assertMsg (((service.certificateFile != null) && (service.certificateKey != null)) || ((service.certificateFile == null) && (service.certificateKey == null))) "Both certificateFile and certificateKey must be set or null on services.tsnsrv.${name}")
```

Comment:
This should be in assertions to give proper module errors

Reply:
That's fantastic, I think I knew about these once but forgot they exist. Much nicer this way.